module_skeleton.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+import rkl
+import math
+import xarray as xr
+from time_utils import datetime_to_float, datetime_from_float
+import pandas as pd
+
+# interferance and clutter removal
+
+# frequency bank of matched filters for a single pulse
+# mf_rx = 2D array (delay, frequency)
+def freq_dft(array1, array2):
+    y = rkl.delay_multiply.delaymult_like_arg2(array2.values, array1.values/np.sum(array1.values), R=1)
+    z = np.fft.fft(y)
+    global f
+    f = np.fft.fftfreq(array1.shape[0], 1e-6)
+    
+    array = np.arange(array2.delay.values[0]-(len(array1)-1), array2.delay.values[0])
+    delay_array = np.append(array, array2.delay.values)
+
+    mf_rx = xr.DataArray(z, coords=dict(t=array2.t.values, delay=('delay', delay_array, {'label': 'Delay (samples)'}), frequency=('frequency', f)), dims=('delay', 'frequency',), name='mf_rx')
```

Comment:
Break this up over multiple lines so it is easy to read without scrolling or wrapping.